modules/history/src/history-helper.js

```diff
@@ -0,0 +1,186 @@
+const SerializableClass = require("abitbol-serializable");
+const self = require("../index.js");
+
+const historyHelper = {
+
+    /**
+     * Check the differences between two state and apply the changes of snapshotState
+     * to currentState
+     * @param  {Object} currentState - The current state of the history
+     * @param  {Object} snapshotState - The new state of the history
+     */
+    applySnapshotDifference: (currentState, snapshotState) => { // eslint-disable-line
+        const {
+            dataStore,
+        } = self.app.modules;
+        // We make a flat object list for each object
+        const currentEntities = historyHelper.flattenSnapshot(currentState);
+        const snapshotEntities = historyHelper.flattenSnapshot(snapshotState);
+
+        Object.keys(currentEntities).forEach(
+            (currentEntityKey) => {
+                let entityFound = false;
+                Object.keys(snapshotEntities).forEach(
+                    (snapshotEntityKey) => {
+                        // Check common entity properties
+                        const currentEntity = currentEntities[currentEntityKey];
+                        const snapshotEntity = snapshotEntities[snapshotEntityKey];
+                        if (currentEntityKey === snapshotEntityKey) {
+                            entityFound = true;
+                            const entity = dataStore.getEntity(currentEntityKey);
+                            // Property is different or doesn't exist
+                            const unserializedEntity = SerializableClass
+                                .$unserialize(snapshotEntity);
+                            Object.keys(snapshotEntity).forEach(
+                                (property) => {
+                                    if ((typeof currentEntity[property] !== "undefined"
+                                        && historyHelper.checkDiff(currentEntity[property],
+                                            snapshotEntity[property]))
+                                        || !currentEntity[property]) {
```

Comment:
`currentEntity[property]` may be set but falsy, like `0` or `""`

Reply:
@nathanidp this part seems weird indeed since  line 36 we already test `typeof currentEntity[property] !== "undefined"` and the checkDiff first thing is to test if property is not null or false. So the second part of the if seems unecessary.
We might want to add a test with some falsy values.